### PR TITLE
Add ability to select loadout on flight creation

### PR DESCRIPTION
**Pull request by Druss99**
These changes allow for users to select a loadout for a flight during flight creation. Tested in retribution UI and checked the created mission to ensure loadouts were properly adjusted.

### CODE REVIEW

**Comment by Druss99** [comment]
This could probably use a few added features like showing the payload when you hover over the name of the loadout or perhaps the full payload preview like on the payload editor page. Dropping this PR in to get an idea of what we might want to fledge out to consider it a full feature or if this is a good start that can be merged in and iterated on in the future.

**Comment by Druss99** [comment]
Added showing the payload when you hover over a loadout and set the default loadout in the drop down based on task selected

File: qt_ui/windows/mission/flight/QFlightCreator.py [hunk 1]
```diff
@@ -85,6 +89,20 @@ def __init__(
...
+        self.loadout_selector.setItemDelegate(LoadoutDelegate(self.loadout_selector))
+        for loadout in Loadout.iter_for_aircraft(self.aircraft_selector.currentData()):
+            self.loadout_selector.addItem(loadout.name, loadout)
+        for loadout in Loadout.default_loadout_names_for(
+            self.task_selector.currentData()
+        ):
+            index = self.loadout_selector.findText(loadout)
+            if index != -1:
+                self.loadout_selector.setCurrentIndex(index)
+                break
```
Comment (Raffson) [inline qt_ui/windows/mission/flight/QFlightCreator.py]: This part could be refactored, along with the same piece from `on_aircraft_changed`, say to a method called `_init_loadout_selector`?

File: qt_ui/windows/mission/flight/QFlightCreator.py [hunk 2]
```diff
@@ -217,8 +236,20 @@ def on_aircraft_changed(self, index: int) -> None:
...
+                self.aircraft_selector.currentData()
+            ):
+                self.loadout_selector.addItem(loadout.name, loadout)
+            for loadout in Loadout.default_loadout_names_for(
+                self.task_selector.currentData()
+            ):
+                index = self.loadout_selector.findText(loadout)
+                if index != -1:
+                    self.loadout_selector.setCurrentIndex(index)
+                    break
```
Comment (Raffson) [inline qt_ui/windows/mission/flight/QFlightCreator.py]: For reference, this is the common part afaict 😅

**Review by Raffson — CHANGES_REQUESTED** [review]

**Comment by Druss99** [comment]
Changed the loadout selector init to a method, tested and all looks good, let me know if I missed anything!

**Review by Raffson — APPROVED** [review]

**Pull request merged**